app/src/main/java/protect/card_locker/ScanActivity.java

```diff
@@ -221,4 +222,21 @@ public void addFromImage(View view) {
             Log.e(TAG, "No activity found to handle intent", e);
         }
     }
+
+    private void onSharedIntent() {
+        Intent intent = getIntent();
+        String receivedAction = intent.getAction();
+        String receivedType = intent.getType();
+
+        if (Intent.ACTION_SEND.equals(receivedAction)) {
+            if (receivedType.startsWith("image/")) {
+                handleActivityResult(Utils.BARCODE_IMPORT_FROM_SHARE_INTENT, RESULT_OK, intent);
+                //Return if barcode wasn't found
+                ScanActivity.this.setResult(RESULT_CANCELED);
+                finish();
+            } else {
+                Log.e(TAG, "Wrong mime-type");
+            }
+        }
+    }
```

Comment:
There is already an `extractIntentFields` function in this class that deals with reading the intent to see what to do. Maybe hook it into there?

Reply:
fixed